### PR TITLE
Add support for default pseudo cluster for chip-repl based yamltests

### PR DESCRIPTION
**Pull request by tehampson**
By supporting basic default pseudo cluster there are more yaml test that start passing. Updating CI to run those test will come in a separate PR as there are a few PRs in flight to modify that file

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: src/controller/python/chip/yaml/runner.py [hunk 1]
```diff
@@ -254,7 +279,9 @@ def __init__(self, test_step):
         args = test_step.arguments['values']
         request_data_as_dict = Converter.convert_list_of_name_value_pair_to_dict(args)
 
-        self._expire_existing_session = request_data_as_dict.get('expireExistingSession', False)
+        # Only if expire existing session is explicitly false do we want to supress expiring
+        # the session.
+        self._expire_existing_session = request_data_as_dict.get('expireExistingSession', True)
```
Comment (bzbarsky-apple) [inline src/controller/python/chip/yaml/runner.py:284]: Wait, why this change?  The comment should explain _why_ we want the behavior, not what the behavior is.

File: src/controller/python/chip/yaml/runner.py [hunk 2]
```diff
@@ -253,8 +278,12 @@ def __init__(self, test_step):
         args = test_step.arguments['values']
         request_data_as_dict = Converter.convert_list_of_name_value_pair_to_dict(args)
 
+        # When expireExistingSession is not provided it is true be default. This is because
+        # expireExistingSession argument was added later on to test a specific CASE situation
+        # where that specific test case did not want existing session to be expired. For more
+        # information see PR #20820.
+        self._expire_existing_session = request_data_as_dict.get('expireExistingSession', True)
```
Comment (bzbarsky-apple) [inline src/controller/python/chip/yaml/runner.py:285]: I think the key part here is that we are preserving the old behavior, and that behavior exists for a reason, and that reason is documented where we currently implement the behavior.

We should copy _that_ comment, explaining the reason, to here, to make it clear why the behavior is what it is.  So replace the comments with some version of https://github.com/project-chip/connectedhomeip/blob/35bd94da455fbab2bf0cc474f220a44c989d32d2/examples/chip-tool/commands/tests/TestCommand.cpp#L40-L42

File: src/controller/python/chip/yaml/runner.py [hunk 3]
```diff
@@ -253,8 +278,13 @@ def __init__(self, test_step):
         args = test_step.arguments['values']
         request_data_as_dict = Converter.convert_list_of_name_value_pair_to_dict(args)
 
+        # There's a chance the commissionee may have rebooted before this call here as part of a
+        # test flow or is just starting out fresh outright. Unless expireExistingSession is
+        # explicitly set, the default behaviour it so make sure we're not re-using any cached CASE
```
Comment (bzbarsky-apple) [inline src/controller/python/chip/yaml/runner.py:283]:
```suggestion
        # explicitly set, the default behaviour is to make sure we're not re-using any cached CASE
```

**Review by bzbarsky-apple — APPROVED** [review]
